dashboard/app/helpers/users_helper.rb

```diff
@@ -168,68 +168,39 @@ def user_summary(user)
       sl.level_ids.each do |level_id|
         # if we have a contained level or BubbleChoice level, use that to represent progress
         level = Level.cache_find(level_id)
-        sublevel_id = level.is_a?(BubbleChoice) ? level.best_result_sublevel(user)&.id : nil
+
         if level.is_a?(BubbleChoice) # we have a parent level
```

Comment:
imo the amount of this function devoted to the special case of `BubbleChoice` -- and the fact that it comes first -- makes this function a lot harder to understand at first glance. this refactor is a _huge_ improvement, but i think it might improve readability significantly if we were to move this whole block into a separate function.

i was initially hesitant to suggest that because passing the top-level `progress` hash and mutating it in a helper didn't sound right, but i realized you could just build a new hash in the helper and merge it with the main one.

Reply:
I toyed with that idea as well, will do.